code/game/gamemodes/meteor/meteors.dm

```diff
@@ -225,12 +231,23 @@ var/global/list/meteors_cataclysm = list(\
 		O.throw_at(dest, 5, 10)
 
 /obj/effect/meteor/proc/meteor_effect()
-	for(var/mob/M in GLOB.player_list)
-		var/turf/T = get_turf(M)
-		if(!T || T.z != src.z)
-			continue
-		var/dist = get_dist(M.loc, src.loc)
-		shake_camera(M, dist > 20 ? 3 : 5, dist > 20 ? 1 : 3)
+	if(flicker_range)
+		for(var/obj/machinery/light/L in range(flicker_range, src))
+			L.flicker(rand(5,15))
+
+	if(shake_range)
+		for(var/mob/M in range(shake_range, src))
+			if(!M.stat && M.client && !istype(M, /mob/living/silicon/ai))\
+				shake_camera(M, 3, 1)
+
+	else
+		for(var/mob/M in GLOB.player_list) //this is an insane way of doing this bay
```

Comment:
_It's not that bad._ 

This list only contains mobs with attached clients; even if our pop doubled, it would be at most 40 iterations, done _once_, over a homogeneous list. Hell, the game does much more every single tick of the health system.

You couldn't even do it by block(), since _all_ mobs on the z-level get _some_ screenshake, and even if you made dist > 20 not do any, you'd have to check a 40x40x1 block of turfs - 160 iterations minimum.

The only real optimization here would be to have a global cache of players **per z-level**, which would remove everything L246 needs to deal with anyway.

Reply:
This whole file is full of some baffling baycode decisions